src/data/matrix/basic.lean

```diff
@@ -1647,148 +1647,151 @@ lemma star_mul [fintype n] [non_unital_semiring α] [star_ring α] (M N : matrix
   star (M ⬝ N) = star N ⬝ star M := conj_transpose_mul _ _
 
 end star
+end matrix
 
 /-- Given maps `(r_reindex : l → m)` and  `(c_reindex : o → n)` reindexing the rows and columns of
-a matrix `M : matrix m n α`, the matrix `M.minor r_reindex c_reindex : matrix l o α` is defined
-by `(M.minor r_reindex c_reindex) i j = M (r_reindex i) (c_reindex j)` for `(i,j) : l × o`.
+a matrix `M : matrix m n α`, the matrix `M.on r_reindex c_reindex : matrix l o α` is defined
+by `(M.on r_reindex c_reindex) i j = M (r_reindex i) (c_reindex j)` for `(i,j) : l × o`.
 Note that the total number of row and columns does not have to be preserved. -/
-def minor (A : matrix m n α) (r_reindex : l → m) (c_reindex : o → n) : matrix l o α :=
-of $ λ i j, A (r_reindex i) (c_reindex j)
+protected def matrix.on (A : matrix m n α) (r_reindex : l → m) (c_reindex : o → n) : matrix l o α :=
+matrix.of $ λ i j, A (r_reindex i) (c_reindex j)
+
+namespace matrix
 
-@[simp] lemma minor_apply (A : matrix m n α) (r_reindex : l → m) (c_reindex : o → n) (i j) :
-  A.minor r_reindex c_reindex i j = A (r_reindex i) (c_reindex j) := rfl
+@[simp] lemma on_apply (A : matrix m n α) (r_reindex : l → m) (c_reindex : o → n) (i j) :
+  A.on r_reindex c_reindex i j = A (r_reindex i) (c_reindex j) := rfl
 
-@[simp] lemma minor_id_id (A : matrix m n α) :
-  A.minor id id = A :=
+@[simp] lemma on_id_id (A : matrix m n α) :
+  A.on id id = A :=
 ext $ λ _ _, rfl
 
-@[simp] lemma minor_minor {l₂ o₂ : Type*} (A : matrix m n α)
+@[simp] lemma on_on {l₂ o₂ : Type*} (A : matrix m n α)
   (r₁ : l → m) (c₁ : o → n) (r₂ : l₂ → l) (c₂ : o₂ → o) :
-  (A.minor r₁ c₁).minor r₂ c₂ = A.minor (r₁ ∘ r₂) (c₁ ∘ c₂) :=
+  (A.on r₁ c₁).on r₂ c₂ = A.on (r₁ ∘ r₂) (c₁ ∘ c₂) :=
 ext $ λ _ _, rfl
 
-@[simp] lemma transpose_minor (A : matrix m n α) (r_reindex : l → m) (c_reindex : o → n) :
-  (A.minor r_reindex c_reindex)ᵀ = Aᵀ.minor c_reindex r_reindex :=
+@[simp] lemma transpose_on (A : matrix m n α) (r_reindex : l → m) (c_reindex : o → n) :
+  (A.on r_reindex c_reindex)ᵀ = Aᵀ.on c_reindex r_reindex :=
 ext $ λ _ _, rfl
 
-@[simp] lemma conj_transpose_minor
+@[simp] lemma conj_transpose_on
   [has_star α] (A : matrix m n α) (r_reindex : l → m) (c_reindex : o → n) :
-  (A.minor r_reindex c_reindex)ᴴ = Aᴴ.minor c_reindex r_reindex :=
+  (A.on r_reindex c_reindex)ᴴ = Aᴴ.on c_reindex r_reindex :=
 ext $ λ _ _, rfl
 
-lemma minor_add [has_add α] (A B : matrix m n α) :
-  ((A + B).minor : (l → m) → (o → n) → matrix l o α) = A.minor + B.minor := rfl
+lemma add_on [has_add α] (A B : matrix m n α) :
+  ((A + B).on : (l → m) → (o → n) → matrix l o α) = A.on + B.on := rfl
```

Comment:
I'm not convinced we want to be reordering the names like this; that wasn't discussed on Zulip.

Reply:
Ok, I've undone the reordering.

Beyond the pure replacement of `minor` by `on`, I have also changed the docstring in `src/linear_algebra/matrix/adjugate.lean` a little because it had the same misunderstanding of "minor" as the rest of mathlib. If you'd like I could put that in a different PR of course.